### PR TITLE
Improve docker down error

### LINKED ISSUE #108: Improve docker down error message
When trying to remove the local deployment when there is no local deployment, the cli returns an error because it can't find the docker-compose file. In this particular case we could give a custom feedback to the user like "no local deployment found, nothing to remove. Use fl admin deploy docker to set one up." or something.

### PR DESCRIPTION
**Pull request by ABrunoRaise**
This PR closes #108 

Specifically:
- adds a `composeList` function to list running compose project
- changes `runShellCmd` input to take out/err buffers as input
- fixes `docker` and `docker_down` tests and mock

Additionally:
- adds newline in `docker_up` output

### CODE REVIEW

File: internal/command/admin/deploy/docker/docker_down_test.go [hunk 1]
```diff
@@ -66,6 +66,46 @@ func TestDockerDownRun(t *testing.T) {
 		require.Error(t, err)
 	})
 
+	t.Run("should return error when unable to find docker compose but local deployment is found", func(t *testing.T) {
```
Comment (mattrent) [inline internal/command/admin/deploy/docker/docker_down_test.go:69]: The message should probably be "[...] when unable to find docker-compose.yml" for better clarity (`docker compose` is technically the command, not the file).

File: internal/command/admin/deploy/docker/docker_down_test.go [hunk 2]
```diff
@@ -66,6 +66,46 @@ func TestDockerDownRun(t *testing.T) {
...
+		out.Reset()
+		mockDockerShell.On("ComposeList").Return([]string{"fl"}, nil).Once()
+
+		err = down.Run(ctx, mockDockerShell, logger)
+		require.Error(t, err)
+		require.Equal(t, err.Error(), "unable to locate docker-compose.yml, but a local deployment was found. The file might have been moved or deleted.")
+
+	})
+
+	t.Run("should return error when unable to find docker compose and local deployment is not found", func(t *testing.T) {
```
Comment (mattrent) [inline internal/command/admin/deploy/docker/docker_down_test.go:89]: Same as the comment on Line 69

**Review by mattrent — CHANGES_REQUESTED** [review]

File: pkg/deploy/docker_test.go [hunk 3]
```diff
@@ -32,6 +33,6 @@ func Test_parseCmd(t *testing.T) {
 }
 
 func Test_runShellCmd(t *testing.T) {
-	err := runShellCmd("echo", "hello")
+	err := runShellCmd(os.Stdout, os.Stderr, "echo", "hello")
```
Comment (giusdp) [inline pkg/deploy/docker_test.go:36]: Now that you've added the out/err buffers we could have better integration tests with the shell. 
For example here you could pass an out buffer and check that the string hello is contained in it.

**Review by giusdp — CHANGES_REQUESTED** [review]

**Comment by giusdp** [comment]
Thanks for contributing @ABrunoRaise ! It looks pretty good, after addressing these few requested changes I'm ready to merge. 

One note for next time: the commits perhaps are way too fine grained with a single file per commit. For instance there is a commit to just add a newline, or the DockerShell mock update could have been added together with the docker.go change, since they are related. If you forget to add something really small, you can do amend a commit with `git commit --amend --no-edit` to add a new change to the previous commit.

Don't worry you don't have to change anything now (we will do a squash merge). If you have multiple small changes which might be related, you can also do one commit with a description under the commit summary to explain more what you've changed (if you use commitizen it's easy to do it).

**Comment by ABrunoRaise** [comment]
I've added the requested changes, let me know if there are any other changes.
Thanks @giusdp @mattrent  for the great tips, I will keep them in mind for future PR.

File: pkg/deploy/docker_test.go [hunk 4]
```diff
@@ -33,6 +34,42 @@ func Test_parseCmd(t *testing.T) {
...
+	t.Run("should return the command output in output buffer", func(t *testing.T) {
+		var outBuf bytes.Buffer
+		err := runShellCmd(&outBuf, os.Stderr, "echo", "hello")
+		assert.Nil(t, err)
+		assert.Equal(t, "hello\n", outBuf.String())
+	})
+
+	t.Run("should return in error buffer all the contents of stderr", func(t *testing.T) {
+		var errBuf bytes.Buffer
+		errSecond := runShellCmd(os.Stdout, &errBuf, "/bin/bash", "-c", "echo hello err 1>&2")
```
Comment (giusdp) [inline pkg/deploy/docker_test.go:47]: Thanks for the changes, they look good @ABrunoRaise. 

One last thing: the new shell tests for the error case use bash directly. They will only work on linux with bash installed. On windows they will fail (and macos too I think as it uses zsh by default). Can you try to change to just /bin/sh and see if they still work as before? At least that should cover macos.

**Review by giusdp — CHANGES_REQUESTED** [review]

File: pkg/deploy/docker_test.go (hunk 4 above)
Comment (ABrunoRaise) [reply pkg/deploy/docker_test.go:47]: Thanks for the feedback, I will test the change.

**Review by mattrent — APPROVED** [review]

**Review by giusdp — APPROVED** [review]

**Pull request merged**